app/org/sagebionetworks/bridge/services/SubpopulationService.java

```diff
@@ -221,7 +221,8 @@ public void deleteSubpopulation(StudyIdentifier studyId, SubpopulationGuid subpo
     public void deleteSubpopulationPermanently(StudyIdentifier studyId, SubpopulationGuid subpopGuid) {
         checkNotNull(studyId);
         checkNotNull(subpopGuid);
-        
+
+        studyConsentService.deleteAllConsentsPermanently(subpopGuid);
```

Comment:
If you do this first, there will be a tiny window where loading the subpopulation might throw an error for lack of a consent... ? Do it after?

Reply:
Good point. Changed.